superset/databases/api.py

```diff
@@ -604,13 +609,8 @@ def test_connection(  # pylint: disable=too-many-return-statements
         # This validates custom Schema with custom validations
         except ValidationError as error:
             return self.response_400(message=error.messages)
-        try:
-            TestConnectionDatabaseCommand(g.user, item).run()
-            return self.response(200, message="OK")
-        except DatabaseTestConnectionFailedError as ex:
-            return self.response_422(message=str(ex))
-        except SupersetErrorException as ex:
-            return self.response(ex.status, message=ex.error.message)
+        TestConnectionDatabaseCommand(g.user, item).run()
+        return self.response(200, message="OK")
```

Comment:
the OpenAPI error responses are not correct now, on the `BaseSupersetModelRestApi` we can declare the new SIP-40 schema https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/api/__init__.py#L296

Reply:
Good catch, I'll fix!